hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueFunction.java

```diff
@@ -41,13 +42,16 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Objects;
+import java.util.function.Function;
 
 public class JsonValueFunction<T> extends VariExpressionWithType<T> implements IdentifiedDataSerializable {
     private static final ILogger LOGGER = Logger.getLogger(JsonValueFunction.class);
     private static final int DEFAULT_ON_EMPTY_OPERAND_INDEX = 2;
     private static final int DEFAULT_ON_ERROR_OPERAND_INDEX = 3;
+    private static final Function<String, JsonPath> COMPILE_FUNCTION = JsonPathUtil::compile;
 
-    private final Cache<String, JsonPath> pathCache = JsonPathUtil.makePathCache();
+    private transient ConcurrentInitialSetCache<String, JsonPath> pathCache;
+    private transient JsonPath constantPathCache;
```

Comment:
The point of `constantPathCache` was to share the compiled result among members, and even cache it in the plan and share it across executions. But that's not the case currently if I'm not mistaken. In the current shape it's very similar to a cache containing a single item.